core/editor/src/tools/mod.rs

```diff
@@ -1,7 +1,9 @@
-use crate::Color;
+use crate::{events::Trace, Color};
 use std::collections::HashMap;
 
 pub struct ToolState {
+	pub mouse_is_clicked: bool,
```

Comment:
Could you just save either a MouseState struct, or a MousKeys to allow to track all mouse buttons?